### PR TITLE
Gwlp 648, Use GWLP_* instead of GWL_* and DWLP_* instead of DWL_*

### PR DESCRIPTION
**Pull request by chcg**
Added patch from http://sourceforge.net/p/notepad-plus/patches/648/ as preparation for x64 builds + some additional cleanups regarding Set/GetWindowLongPtr

### CODE REVIEW

File: PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp [hunk 1]
```diff
@@ -255,19 +255,9 @@ void ScintillaEditView::init(HINSTANCE hInst, HWND hPere)
 	
 	_codepage = ::GetACP();
 
-	//Use either Unicode or ANSI setwindowlong, depending on environment
```
Comment (NN---) [inline PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp]: Why do you think this code is not needed anymore ?

Comment (chcg) [inline PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp]: Think this part is a leftover of the old days with ANSI and Unicode version (https://notepad-plus-plus.org/download/v5.9.6.2.html) , but 

_callWindowProc = CallWindowProcW;
should be
_callWindowProc = CallWindowProc;

**Comment by chcg** [comment]
"Code enhancement" was just the name used on donho:master to sync this patch with the current master revision

**Comment by Croydon** [comment]
@chcg Well, my comment was addressed to @donho. I did directly comment his commit, but GitHub is displaying all comments on all commits ;)

**Comment by ariccio** [comment]
> I did directly comment his commit, but GitHub is displaying all comments on all commits ;)

Hmm. Yes, this is a bit confusing >:(

**Comment by NN---** [comment]
It has now conflicts.
Please update the PR.

**Comment by NN---** [comment]
Merge conflicts :(

**Comment by NN---** [comment]
I think this should move to milestone 7 with the #178

**Pull request merged**